packages/browser/data-table/InternalUncontrolledDataTable/hooks/useUncontrolledDataTable.ts

```diff
@@ -75,15 +75,27 @@ export const useUncontrolledDataTable = <T>({
     setSearchParams(searchParams);
     syncData(searchParams);
   };
+  const [keyword, setKeyword] = useState(
+    searchParams.get(UncontrolledDataTableURLParams.KEYWORD)
+  );
 
-  const onKeywordChange = (keyword: string) => {
-    if (keyword.trim())
-      searchParams.set(UncontrolledDataTableURLParams.KEYWORD, keyword);
+  const handleKeywordDebounced = (debouncedKeyword: string) => {
+    if (debouncedKeyword.trim())
+      searchParams.set(
+        UncontrolledDataTableURLParams.KEYWORD,
+        debouncedKeyword
+      );
     else searchParams.delete(UncontrolledDataTableURLParams.KEYWORD);
     setSearchParams(searchParams);
     syncData(searchParams);
   };
 
+  const debounceValue = useDebounce(handleKeywordDebounced);
```

Comment:
`debounceValue` can be changed to a better name (what value are we talking about)